content/en/docs/reference/plugins/plugin-api-reference.md

```diff
@@ -691,6 +783,52 @@ typedef struct
 } ss_plugin_init_tables_input;
 ```
 
+### Obtaining subtables accessors
+
+Obtaining table accessors is a bit different for subtables. 
+
+Table accessors can be only obtained at initialization time, however tables may be empty at this time, which means subtables may be not existing yet.
```

Comment:
```suggestion
Table accessors can be only obtained at initialization time, however tables may be empty at this time, which means subtables may be not yet exist.
```

Reply:
Could you provide clear examples how to use the new API instead of just providing the source code snippets?

For example how to access the fd table fields and the cmdline args and how is it different from accessing the currently available fields (see below)?

```
Header:
falcosecurity::table_field m_comm;

CPP:
init(falcosecurity::init_input& in)
{
    using st = falcosecurity::state_value_type;
    auto& t = in.tables();
    m_thread_table = t.get_table(THREAD_TABLE_NAME, st::SS_PLUGIN_ST_INT64);
    m_comm = m_thread_table.get_field(t.fields(), "comm", st::SS_PLUGIN_ST_STRING);
...

std::string tstr;
m_comm.read_value(tr, thread_entry, tstr);
```

Reply:
There is already an example that shows how to obtain subtable accessors.
Once you obtained accessors and subtables handles, accessing subtables fields works just like any other regular table.
But I agree on making this more clear by providing another example 👍🏻

Reply:
It may seem intuitive to the folks who developed it. For end users who seek to develop a plugin the docs are not sufficient IMO. Instead of source code snippets from libs, what would be better would be clear examples on how to use the API similar to the snippet above.

Reply:
I agree with @incertum . 

Just a question: do we want to address this point in this PR? Or do we want to unblock this PR and enhance its content in a follow-up PR? :thinking:

Reply:
The last commit already adds [an example](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#accessing-subtables) that shows how to access subtable fields. 
There is also [this example](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#obtaining-subtables-accessors) that shows how to obtain subtable accessors. 
These two examples are what the user needs to understand how to access subtables IMHO.

Please tell me if I am missing something 🤔

Reply:
LGTM. 

@incertum ?

Reply:
Hm the links seem to show libs snippet again? And does not appear to be an example of how to use it in a real plugin developed with the plugin SDK in the plugins repo.

Given the below example that shows how I read the `comm` value how do I read `m_args`, and `m_fd_name` values?

```
Header:
falcosecurity::table_field m_comm;

CPP:
init(falcosecurity::init_input& in)
{
    using st = falcosecurity::state_value_type;
    auto& t = in.tables();
    m_thread_table = t.get_table(THREAD_TABLE_NAME, st::SS_PLUGIN_ST_INT64);
    m_comm = m_thread_table.get_field(t.fields(), "comm", st::SS_PLUGIN_ST_STRING);
...

std::string tstr;
m_comm.read_value(tr, thread_entry, tstr);
```

Reply:
@incertum 
🤔 I think you are confusing plugin examples written using the plugin API, with libs snippets.
The purpose of the `Falco Plugins API Reference` page is to document symbols/types of the plugin API which is a plain-C interface.
For this reason, the example I provided shows how to use the plugin API directly without using the C++/Go plugin SDK.

You mentioned accessing the `fd_name` field. The example I provided shows exactly how to do that: [here](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#obtaining-subtables-accessors) it shows how to get the field accessor to that field, and [here](https://github.com/mrgian/falco-website/blob/plugin-api-docs-3.5.0/content/en/docs/reference/plugins/plugin-api-reference.md#accessing-subtables) it shows how to use that accessor to obtain the actual `fd_name` value from one of the subtables.

Again, this example shows how to use the the plugin API and not the plugin SDK.
Documenting the plugin SDK is out of the scope of this website page IMHO. 
Also support for subtables in the C++ plugin SDK has not been merged yet (https://github.com/falcosecurity/plugin-sdk-cpp/pull/38) 

From what I know, the C++ SDK is still an experimental project, it can be subject to breaking changes and it has not been released. For this reason it has no documentation yet, however it will provide [examples](https://github.com/falcosecurity/plugin-sdk-cpp/pull/38/files#diff-c408a815df00653669c8ace735b01a6c91329e5537ab0bf32ca7dbbe8d17d15eR140).

Reply:
Hey I am more interested in how to actually use it in a plugin, I added bunch of comments to my open PR https://github.com/falcosecurity/plugins/pull/419#discussion_r1649282173. Appreciate clear examples once the SDK is updated.

Reply:
I'll provide some examples when the C++ SDK is ready :)
But I think it should be addressed in another PR, since the goal of this one is updating the plugin API reference.

Reply:
Ok, I'd like to defer this to a follow-up PR and unblock this meantime.